### PR TITLE
Prevent logging changesets on the console for Quarkus

### LINKED ISSUE #13126: When running JPA Map liquibase changeset, don't log the on the console
### Describe the bug

This happens for the Quarkus distribution, and logging happend via java.util.logging, outside the other loggers. 

This has been solved for the legacy JPA implementation by creating a Liquibase Scope

### Version

999-SNAPSHOT

### Expected behavior

Don't log changesets on the console

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

Run the Quarkus distribution with an empty database and storage jpa enabled.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #13126

### CODE REVIEW

**Comment by ahus1** [comment]
To test this, run Quarkus with an empty database. See #9596 how to run JPA Map storage with Quarkus.

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/liquibase/connection/DefaultLiquibaseConnectionProvider.java [hunk 1]
```diff
@@ -56,14 +59,50 @@ public void close() {
     public Liquibase getLiquibaseForCustomUpdate(final Connection connection, final String defaultSchema, final String changelogLocation,
                                                  final ClassLoader classloader, final String changelogTableName) throws LiquibaseException {
 
-        Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnection(connection));
-        if (defaultSchema != null) {
-            database.setDefaultSchemaName(defaultSchema);
+        String scopeId = enterLiquibaseScope();
+        try {
+            Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnectionFromPool(connection));
```
Comment (sguilhen) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/liquibase/connection/DefaultLiquibaseConnectionProvider.java:64]: Shouldn't we also wrap the connection [here](https://github.com/keycloak/keycloak/blob/main/model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/liquibase/updater/MapJpaLiquibaseUpdaterProvider.java#L201)?

**Comment by sguilhen** [comment]
@ahus1 Thanks for looking into this. The changes themselves look fine to me, only had a comment about the possibility to wrap the connection in another location.

**Comment by ahus1** [comment]
@sguilhen - thanks, changed that. Now it's doing a better job in housekeeping. Builds are still running, then it's ready for another review.

**Review by DGuhr — APPROVED** [review]
From a distribution side these changes lgtm! Thanks @ahus1 :)

 Follow-up note/question:
@ahus1 
I started the server using a clean postgres db
```
docker run --rm --name kctd-postgres -p 5432:5432 -e POSTGRES_USER=keycloak -e POSTGRES_PASSWORD=keycloak -d postgres   
```
and this command:
```
./kc.sh start-dev --storage=jpa --db-username=keycloak --db-password=keycloak 
```
all fine, liquibase is now using a proper logger, good work 👍 

But after creating the initial admin user and logging into the admin console, I created a new realm "foobar" and clicked through the menu to see what changeset gets applied when (old / default admin UI). 

When I clicked on "groups" the first time, the following exception occured:
```
2022-07-15 14:06:04,965 WARN  [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-2) SQL Error: 0, SQLState: 42P01
2022-07-15 14:06:04,965 ERROR [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-2) ERROR: relation "kc_group" does not exist
  Position: 215
2022-07-15 14:06:04,971 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-2) Uncaught server error: org.hibernate.exception.SQLGrammarException: could not extract ResultSet
	at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:103)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:37)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:67)
	at org.hibernate.loader.Loader.getResultSet(Loader.java:2322)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2075)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2037)
	at org.hibernate.loader.Loader.scroll(Loader.java:2945)
	at org.hibernate.loader.hql.QueryLoader.scroll(QueryLoader.java:610)
	at org.hibernate.hql.internal.ast.QueryTranslatorImpl.scroll(QueryTranslatorImpl.java:452)
	at org.hibernate.engine.query.spi.HQLQueryPlan.performScroll(HQLQueryPlan.java:353)
	at org.hibernate.internal.SessionImpl.scroll(SessionImpl.java:1596)
	at org.hibernate.query.internal.AbstractProducedQuery.doScroll(AbstractProducedQuery.java:1588)
	at org.hibernate.query.internal.AbstractProducedQuery.scroll(AbstractProducedQuery.java:1574)
	at org.hibernate.query.internal.AbstractProducedQuery.stream(AbstractProducedQuery.java:1598)
	at org.hibernate.query.criteria.internal.compile.CriteriaQueryTypeQueryAdapter.stream(CriteriaQueryTypeQueryAdapter.java:91)
	at org.hibernate.query.Query.getResultStream(Query.java:1140)
	at org.keycloak.models.map.storage.jpa.JpaMapKeycloakTransaction.read(JpaMapKeycloakTransaction.java:167)
	at org.keycloak.models.map.group.MapGroupProvider.getGroupsStreamInternal(MapGroupProvider.java:104)
	at org.keycloak.models.map.group.MapGroupProvider.getTopLevelGroupsStream(MapGroupProvider.java:162)
	at org.keycloak.models.map.realm.MapRealmAdapter.getTopLevelGroupsStream(MapRealmAdapter.java:1424)
	at org.keycloak.models.utils.ModelToRepresentation.toGroupHierarchy(ModelToRepresentation.java:169)
	at org.keycloak.services.resources.admin.GroupsResource.getGroups(GroupsResource.java:86)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:67)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:55)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$5.handle(VertxHttpRecorder.java:380)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$5.handle(VertxHttpRecorder.java:358)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$1(QuarkusRequestFilter.java:71)
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:545)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.postgresql.util.PSQLException: ERROR: relation "kc_group" does not exist
  Position: 215
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:355)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:490)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:408)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:166)
	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:118)
	at io.agroal.pool.wrapper.PreparedStatementWrapper.executeQuery(PreparedStatementWrapper.java:78)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:57)
	... 72 more

2022-07-15 14:06:04,987 INFO  [org.keycloak.models.map.storage.jpa.liquibase.updater.MapJpaLiquibaseUpdaterProvider] (executor-thread-4) Updating database. Using changelog META-INF/jpa-groups-changelog.xml
```
as you can see from the log, it seems the changeset gets applied *after* keycloak tries to access the groups table, blindly guessing: these are run on different threads, not seeing each other. 

Is this a known bug? maybe I started kc wrong? classical layer 8?
 If not, I can create an issue if you want. 
I was still able to access the groups page, and also able to create a new group, and switching back to other settings and then to group the exception was gone at the second time.

**Comment by ahus1** [comment]
@DGuhr - yes, I've seen that exception as well. It has been around for some time, but I didn't have the time to investigate it, yet. 

It is unrelated to this change. I might be able to fix it as part of #13130. I'll add a note to that ticket. If it is not being fixed there, it will get its own issue.

**Comment by DGuhr** [comment]
@ahus1 np, just wanted to make you aware :)

**Comment by sguilhen** [comment]
> @DGuhr - yes, I've seen that exception as well. It has been around for some time, but I didn't have the time to investigate it, yet.

Yeah, I can confirm this has been around for some time.

**Review by sguilhen — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving per @sguilhen and @DGuhr 's approvals

**Comment by hmlnarik** [comment]
Model test failure is unrelated

**Pull request merged**